src/transformers/trainer.py

```diff
@@ -2164,7 +2174,7 @@ def evaluation_loop(
 
         # if full fp16 is wanted on eval and this ``evaluation`` or ``predict`` isn't called while
         # ``train`` is running, halve it first and then put on device
-        if not self.is_in_train and self.args.fp16_full_eval:
+        if not self.is_in_train and self.args.half_precision_full_eval:
```

Comment:
So if we name the new var "mixed_half_precision" - this one won't work anymore. as it can't be "half_precision_full"

If we name it "half_precision_mixed" then it works.

As we have 2 modes:

- mixed half-precision - AMP
- full half-precision - `.half` / `.to(bf16)`

I suppose the latter sounds confusing as well - full half is a bit of an oxymoron here.

Reply:
What about simply `half_precision_eval`?

Reply:
Perhaps then:

1. `mixed_half_precision_backend` - (instead of the current `fp16_backend`)
2. `half_precision_eval` -  (instead of the current `fp16_full_eval`)

On the other hand AMP = automatic mixed precision, so there is no half in there, so may be these 2 then?

1. `mixed_precision_backend` - (instead of the current `fp16_backend`)
2. `half_precision_eval`-  (instead of the current `fp16_full_eval`) 

It'd be nice to find something less long to type, but a shorter `mp_` would be too confusing because of model parallel, why not just use `amp`?

1. `amp_backend` - (instead of the current `fp16_backend`)
2. `half_precision_eval`-  (instead of the current `fp16_full_eval`)

Reply:
I agree with the last suggestion for naming.